.pipelines/templates/ToolchainBuild.yml

```
@@ -54,3 +54,31 @@ steps:
       tar -C "${{ parameters.buildRepoRoot }}/build/logs/toolchain" -czf "$published_logs_dir/toolchain.logs.tar.gz" .
     condition: always()
     displayName: "Copy logs for publishing"
+
+  - bash: |
+      # Publish the list of packages that should be re-tested during full package build.
+      #   This list will be the contents of 'built_specs_list.txt' in the toolchain build logs directory, but only for
+      #   packages that have a '%check' section in their spec file. The assumption is that all packages will have a 
+      #   '%check' section for all architectures, or none.
+      built_list="${{ parameters.buildRepoRoot }}"/build/logs/toolchain/built_specs_list.txt
+      specs_dir="${{ parameters.buildRepoRoot }}/SPECS/"
+      if [[ -f "$built_list" ]]; then
+        built_specs=$(xargs < "$built_list")
+        retest_list=()
+        for s in ${built_specs}; do
+          if [ ! -f "${specs_dir}/${s}/${s}".spec ]; then
+              echo "ERROR: '${specs_dir}/${s}/${s}.spec' does not exist"
```

An idea to make the errors even easier to spot:
```suggestion
              echo "##[error]ERROR: '${specs_dir}/${s}/${s}.spec' does not exist"
```

This will colour the message red in the pipeline's output.